publication-rest/src/test/java/no/unit/nva/publication/PublicationResponseFactoryTest.java

```diff
@@ -0,0 +1,129 @@
+package no.unit.nva.publication;
+
+import static no.unit.nva.commons.json.JsonUtils.dtoObjectMapper;
+import static no.unit.nva.model.testing.PublicationGenerator.randomPublication;
+import static no.unit.nva.model.testing.associatedartifacts.AssociatedArtifactsGenerator.randomAssociatedLink;
+import static no.unit.nva.model.testing.associatedartifacts.AssociatedArtifactsGenerator.randomOpenFile;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.List;
+import java.util.Set;
+import no.unit.nva.api.PublicationResponse;
+import no.unit.nva.clients.IdentityServiceClient;
+import no.unit.nva.model.FileOperation;
+import no.unit.nva.model.associatedartifacts.NullAssociatedArtifact;
+import no.unit.nva.model.associatedartifacts.file.FileDto;
+import no.unit.nva.publication.model.business.Resource;
+import nva.commons.apigateway.RequestInfo;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class PublicationResponseFactoryTest {
+
+    @Test
+    void associatedLinkShouldHaveTypeWhenSerializedThroughResponse() throws JsonProcessingException {
```

Comment:
Suggestion: Could you split the statements here for readability? Also, `var publicaton` is actually a `Resource`?


Example:

```
var publication = randomPublication().copy().withAssociatedArtifacts(List.of(new NullAssociatedArtifact())).build();
var resource = Resource.fromPublication(publication)

var response = PublicationResponseFactory.create(publication, getRequestInfo(), getIdentityServiceClient());
var jsonString = dtoObjectMapper.writeValueAsString(response);
var typeCount = StringUtils.countMatches(actualString, "AssociatedLink\"")

assertEquals(1, typeCount)
```